centinel/experiments/baseline_linear.py

```diff
@@ -227,23 +236,28 @@ def run_file(self, input_file):
                 pcap_results[pcap_indexes[url]] = td.pcap()
 
             # Meta-data
-
-            # if meta is a pair of comma-separated values,
-            # they should be treated as country and category
-            if len(meta) == 2:
-                country = meta[0].strip().upper()
-                category = meta[1].strip().upper()
-
-                meta = { "country" : country,
-                         "category" : category
-                       }
-
             url_metadata_results[url] = meta
 
         result["http"] = http_results
         result["tls"] = tls_results
         result["dns"] = dns_results
         result["traceroute"] = traceroute_results
+
+        # if we have an index row, we should turn URL metadata
```

Comment:
you can just have csv.DictWriter create a dictionary for each row, then you could make a new dictionary where the keys are urls and the values are the rows from the csv reader. This would let you do everything in one pass, which would be faster and less code.